src/main/kotlin/org/magmaoffenburg/roboviz/gui/MainWindow.kt

```diff
@@ -0,0 +1,171 @@
+import org.magmaoffenburg.roboviz.gui.menus.*
+import org.magmaoffenburg.roboviz.gui.windows.LogPlayerControlsPanel
+import org.magmaoffenburg.roboviz.rendering.Renderer
+import org.magmaoffenburg.roboviz.util.DataTypes
+import org.magmaoffenburg.roboviz.util.SwingUtils
+import rv.comm.rcssserver.ServerComm
+import java.awt.BorderLayout
+import java.awt.Dimension
+import java.awt.GraphicsEnvironment
+import java.awt.Point
+import java.awt.event.KeyEvent
+import java.awt.event.WindowAdapter
+import java.awt.event.WindowEvent
+import javax.imageio.ImageIO
+import javax.swing.JFrame
+import javax.swing.JMenuBar
+import javax.swing.SwingUtilities
+
+/**
+ * FIXME auto focus for canvas to allow key listener to work
+ */
+class MainWindow : JFrame(), ServerComm.ServerChangeListener {
+
+    private val windowTitle = "${Main.name} ${Main.version}"
+
+    private var isFullscreen = false
+
+    companion object {
+        lateinit var instance: MainWindow
+        lateinit var glCanvas: GLCanvas
+
+        var logPlayerControls: LogPlayerControlsPanel? = null
+    }
+
+    init {
+        initializeWindow()
+        initializeMenu()
+        initializeGLCanvas()
+
+        isVisible = true // set visible after everything is initialized
+        instance = this
+
+        if (Main.mode == DataTypes.Mode.LOG) {
+            initializeLogPlayerControls()
+        }
+    }
+
+    private fun initializeWindow() {
+        title = windowTitle
+        defaultCloseOperation = EXIT_ON_CLOSE
+        size = Dimension(Graphics.frameWidth, Graphics.frameHeight)
+        jMenuBar = JMenuBar()
+        iconImage = ImageIO.read(MainWindow::class.java.getResource("/images/icon.png"))
+
+        // set the main window position
+        location = if (Graphics.centerFrame) {
+            SwingUtils.centerWindowOnScreen(this, Point(0,0))
+        } else {
+            Point(Graphics.frameX, Graphics.frameY)
+        }
+        toFront()
+
+        addWindowListener(object : WindowAdapter() {
+            override fun windowClosing(e: WindowEvent?) {
+                Graphics.frameWidth = width
+                Graphics.frameHeight = height
+
+                Main.config.write()
+                Renderer.instance.dispose(glCanvas)
+            }
+        })
+    }
+
+    private fun initializeMenu() {
+        jMenuBar.removeAll()
+
+        when (Main.mode) {
+            DataTypes.Mode.LIVE -> initializeLiveMenu()
+            DataTypes.Mode.LOG -> initializeLogMenu()
+        }
+    }
+
+    private fun initializeLiveMenu() {
+        val connection = ConnectionMenu(this)
+        val server = ServerMenu()
+        val view = ViewMenu()
+        val camera = CameraMenu()
+        val help = HelpMenu()
+
+        // Live Mode specific items
+        view.addSeparator()
+        view.addItem("Toggle Server Speed", KeyEvent.VK_M) { println("TODO toggleShowServerSpeed") }
+        view.addItem("Playmode Overlay", KeyEvent.VK_O) { println("TODO openPlaymodeOverlay") }
```

Comment:
These two menu items don't work yet. However, the keystroke does work because the key press event is also handled in LiveGameScreen. I'm not sure how this was handled before the Kotlin rewrite, but this causes the other events accessible from the menu being executed twice (which does not sound good to me). At least when executed through a keystroke...

Reply:
I think we should remove key event handling from LiveGameScreen, as it is mostly handled by the Menu/MenuActions class.